src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject_IIntellisenseBuildTarget.cs

```diff
@@ -0,0 +1,129 @@
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Build.Execution;
+using Microsoft.Build.Logging;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    internal partial class AbstractProject : IIntellisenseBuildTarget
+    {
+        private static readonly object s_diagnosticKey = new object();
+        private static readonly FieldInfo s_fieldInfo = typeof(BuildManager).GetField("_buildParameters", BindingFlags.GetField | BindingFlags.Instance | BindingFlags.NonPublic);
+
+        // REVIEW: can temp path changed while process running?
+        private static readonly string GeneralLogFile = $"{Path.GetTempPath()}\\<guid>.designtime.log";
+
+        // set default to true so that we maintain old behavior when project system doesn't
+        // implement IIntellisenseBuildTarget
+        private bool _isIntellisenseBuildSucceeded = true;
+        private string _intellisenseBuildFailureReason = null;
+
+        public void SetIntellisenseBuildResult(bool succeeded, string reason)
+        {
+            // set intellisense related info
+            _isIntellisenseBuildSucceeded = succeeded;
+            _intellisenseBuildFailureReason = string.IsNullOrWhiteSpace(reason) ? null : reason.Trim();
+
+            UpdateHostDiagnostics(succeeded, reason);
+
+            if (_pushingChangesToWorkspaceHosts)
+            {
+                // set workspace reference info
+                ProjectTracker.NotifyWorkspaceHosts(host => (host as IVisualStudioWorkspaceHost2)?.OnIsComplete(Id, succeeded));
+            }
+        }
+
+        private void UpdateHostDiagnostics(bool succeeded, string reason)
+        {
+            if (!succeeded)
+            {
+                // report intellisense build failure to error list
+                this.HostDiagnosticUpdateSource?.UpdateDiagnosticsForProject(
+                    _id, s_diagnosticKey, SpecializedCollections.SingletonEnumerable(CreateIntellisenseBuildFailureDiagnostic(reason)));
+            }
+            else
+            {
+                // clear intellisense build failure diagnostic from error list.
+                this.HostDiagnosticUpdateSource?.ClearDiagnosticsForProject(_id, s_diagnosticKey);
+            }
+        }
+
+        private DiagnosticData CreateIntellisenseBuildFailureDiagnostic(string reason)
+        {
+            // log intellisense build failure
+            Logger.Log(FunctionId.IntellisenseBuild_Failed, KeyValueLogMessage.Create(m => m["Reason"] = reason ?? string.Empty));
+
+            var isEnabledByDefault = true;
```

Comment:
You do already.  You use named arguments for 'title' and 'description'.  The only one you'd have to add is 'id'.